sdks/java/core/src/main/java/org/apache/beam/sdk/io/CountingInput.java

```diff
@@ -75,17 +75,21 @@
    * from {@code 0} to {@code numElements - 1}.
    */
   public static BoundedCountingInput upTo(long numElements) {
-    checkArgument(numElements > 0, "numElements (%s) must be greater than 0", numElements);
+    checkArgument(numElements >= 0,
+        "numElements (%s) must be greater than or equal to 0",
+        numElements);
     return new BoundedCountingInput(numElements);
   }
 
   /**
    * Creates a {@link BoundedCountingInput} that will produce elements
    * starting from {@code startIndex} to {@code endIndex - 1}.
+   * If {@code startIndex == endIndex}, then no elements will be produced.
```

Comment:
what about:
"starting from {@code startIndex} (inclusive) to {@code endIndex} (exclusive)."